test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,167 @@
+		BeforeEach(func() {
+			uninstallDefaultNMState()
+		})
+		It("should uninstall handler and webhook", func() {
+			Eventually(func() bool {
+				_, err := getDefaultNMStateHandler()
+				return apierrors.IsNotFound(err)
+			}, 60*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Daemon set for NMState should be removed, but is not"))
+			Eventually(func() bool {
+				_, err := getDefaultNMStateWebhook()
+				return apierrors.IsNotFound(err)
+			}, 60*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Deployment for NMState webhook should be removed, but is not"))
+		})
+	})
+})
+
+func getDefaultNMState() nmstatev1beta1.NMState {
+	return nmstatev1beta1.NMState{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nmstate",
+			Namespace: "nmstate",
+		},
+	}
+}
+
+func getDefaultHandler() v1.DaemonSet {
+	return v1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nmstate-handler",
+			Namespace: "nmstate",
+		},
+	}
+}
+
+func getDefaultWebhook() v1.Deployment {
+	return v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nmstate-webhook",
+			Namespace: "nmstate",
+		},
+	}
+}
+
+func getNMStateWrongName() nmstatev1beta1.NMState {
+	nmstate := getDefaultNMState()
+	nmstate.ObjectMeta.Name = "nmstate-wrong"
+	return nmstate
+}
+
+func installNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeTrue(), "NMState CR does not exist yet")
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR created without error")
+}
+
+func installDefaultNMState() {
+	installNMState(getDefaultNMState())
+}
+
+func uninstallNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeFalse(), "NMState CR exists")
+	err = framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR successfully removed")
+}
+
+func uninstallDefaultNMState() {
+	uninstallNMState(getDefaultNMState())
+}
+
+func getNMStateOperator(nmstate nmstatev1beta1.NMState) (nmstatev1beta1.NMState, error) {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Namespace: nmstate.ObjectMeta.Namespace, Name: nmstate.ObjectMeta.Name}, &nmstate)
+	return nmstate, err
+}
+
+func getDefaultNMStateOperator() (nmstatev1beta1.NMState, error) {
+	return getNMStateOperator(getDefaultNMState())
+}
+
+func getNMStateHandler(handler v1.DaemonSet) (v1.DaemonSet, error) {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Namespace: handler.ObjectMeta.Namespace, Name: handler.ObjectMeta.Name}, &handler)
+	return handler, err
+}
+
+func getDefaultNMStateHandler() (v1.DaemonSet, error) {
+	return getNMStateHandler(getDefaultHandler())
+}
+
+func getNMStateWebhook(webhook v1.Deployment) (v1.Deployment, error) {
```

Comment:
call this getDeployment and remove all reference to handler since this is generic, move this to test/e2e/deployment/